synapse/storage/databases/main/deviceinbox.py

```diff
@@ -639,9 +645,9 @@ def reindex_txn(conn):
     async def _remove_deleted_devices_from_device_inbox(
         self, progress: JsonDict, batch_size: int
     ) -> int:
-        """A background update that deletes all device_inboxes for deleted devices.
+        """No-op.
```

Comment:
better to use `register_noop_background_update` for this. Stick the "Used to be a background update that deletes all device_inboxes for deleted devices" comment next to the call to `register_noop_background_update`

Reply:
Ah, I forgot `register_noop_background_update` was a thing